draft-ietf-tls-esni.md

```diff
@@ -476,9 +475,10 @@ it does a standard ClientHello, with the exception of the following rules:
 1. It MUST NOT include the "pre_shared_key" extension. (See
    {{flow-clienthello-malleability}}.)
 
-The client might duplicate non-sensitive extensions in both messages. However,
-implementations need to take care to ensure that sensitive extensions are not
-offered in the ClientHelloOuter. See {{outer-clienthello}} for additional
+Next, the client constructs EncodedClientHelloInner as described in
```

Comment:
Between #330 and #335, I spent a bit of time trying to make sure the order here was consistent throughout the document. I don't particularly care what the order is, just that it's consistent. If we change it, we'll need to adjust several things in tandem:

* Line 462 ("If it compressed any extensions in [...]") is written assuming the other order. That'll want to read more like "If it intends to compress any extensions [...]" from line 455. 

* {{encoding-inner}} itself also needs to change. "When sending ClientHello, the client first [...] It then computes a new value, the EncodedClientHelloInner, by [...]" and "The client then MAY substitute extensions which it knows will be duplicated in ClientHelloOuter".

Reply:
Reverted. (Apologies for stepping on your toes here, I had forgotten there was a particular order.)